src/createScreen/renderTypes.tsx

```diff
@@ -0,0 +1,31 @@
+import Button from '../components/button.tsx';
+import createPlayer from '../components/hoc/createPlayerInfo.tsx';
+import { useContext } from 'react';
+import PlayerContext from '../components/PlayerContext.tsx'
+import { Link } from "react-router-dom";
+const heroesTypes = ["Mage", "Melee", "Range"];
+interface RenderTypesProps {
+    playerName: string;
+}
+
+export default function RenderTypes({ playerName }: RenderTypesProps) {
+    const { setCurrentPlayer } = useContext(PlayerContext);
+
+    if (!setCurrentPlayer) {
+        return <div>Error: setCurrentPlayer is not available.</div>;
+    }
```

Comment:
why do you need this check for setCurrentPlayer?

Reply:
Cannot dispatch an element that is possibly undefined